cdap-proto/src/main/java/co/cask/cdap/proto/RunRecord.java

```diff
@@ -44,14 +46,25 @@
   @SerializedName("adapter")
   private final String adapterName;
 
+  @Nullable
+  @SerializedName("properties")
+  private final Map<String, String> properties;
+
   public RunRecord(String pid, long startTs, @Nullable Long stopTs, ProgramRunStatus status,
-                   @Nullable String adapterName, @Nullable String twillRunId) {
+                   @Nullable String adapterName, @Nullable String twillRunId,
+                   @Nullable Map<String, String> properties) {
```

Comment:
It seems better to require that properties is not null, but rather an empty map if there is nothing there. That would be consistent with what we do for other similar parameters, such as RuntimeArguments or DatasetProperties.